### DISCUSSION TITLE
Managing Local Accounts using ArgoCD Operator

### DISCUSSION THREAD

**Discussion by jritter** [argoproj-labs/argocd-operator#551] [Q&A]
Hi folks,

Am I missing something or is it not possible at the moment to manage local accounts in ArgoCD [1] when using ArgoCD Operator?

We are using Dex to integrate with the OpenShift authentication, but this is not an ideal solution for automation. I'm looking for an elegant way to create local accounts and a token which could be used for automation through the API.

Any idea?

-- Juerg

[1] https://argo-cd.readthedocs.io/en/stable/operator-manual/user-management/

**Comment by iam-veeramalla** [comment]
Hi @jritter , Argo CD supports local users but it is not recommended. 
```
Additional users for a very small team where use of SSO integration might be considered an overkill. The local users don't provide advanced features such as groups, login history etc. So if you need such features it is strongly recommended to use SSO.
```

More details here:
https://argo-cd.readthedocs.io/en/stable/operator-manual/user-management/

**Reply by jritter** [reply]
Hi @iam-veeramalla ,

Thanks for your answer. I have seen this page, and it also says:

"The local users/accounts feature serves two main use-cases:

Auth tokens for Argo CD management automation. It is possible to configure an API account with limited permissions and generate an authentication token. Such token can be used to automatically create applications, projects etc."

This is what I'm looking for, additional to the SSO solution that we are using (OpenShift OAuth) I would like to create an account which I can use for automation purposes. And while I'm clear on how this can be achieved in ArgoCD, I'm looking for a solution to have that happen if the ArgoCD instance is managed by the ArgoCD operator.

**Reply by iam-veeramalla** [reply]
Got it !! So you are looking for a place holder in the Argo CD CR to create Argo CD users ?

**Reply by jritter** [reply]
Exactly! Define a service account and assign a role to it within the ArgoCD CR, that's exactly what I'm looking for.

**Reply by iam-veeramalla** [reply]
I remember we had a similar discussion before. What do you think about this support @jannfis ?

**Reply by sylus** [reply]
Just was curious if we still were looking at adding this?

**Reply by herzogf** [reply]
We stumbled upon the same issue and I just wanted to point to https://github.com/redhat-developer/gitops-operator/issues/370#issuecomment-1233166267 and its suggestion to use the extraConfig field to add additional entries (e.g. user accounts) to the argocd-cm:
https://argocd-operator.readthedocs.io/en/latest/usage/extra-config/
